### ISSUE TITLE
[FEAT] Add Attention Visualization for Vision Transformers (ViT/DINOv2) and ConvNets

### ISSUE THREAD

**Issue by blackpearl006** [lightly-ai/lightly-train#333]
### 💡 Is your feature request related to a problem?

When continuing pretraining DINOv2 on domain-specific datasets (e.g., medical imaging), there's no built-in way to visualize what the model is learning. Without attention visualization, it's difficult to debug whether the model focuses on meaningful regions or validate that domain adaptation is working correctly.

### 🧰 Describe the solution you'd like

Add built-in attention visualization support for Vision Transformers and ConvNets:

**For ViT/DINOv2:**
- Visualize attention maps from the last ViT block's CLS token (default)
- Save attention heatmaps overlaid on original images during training
- Log visualizations to WandB/TensorBoard

**For ConvNets:**
- Integrated Gradients or GradCAM

**Proposed API:**
```python
from lightly.utils.visualization import AttentionVisualizer

visualizer = AttentionVisualizer(
    model=model.backbone,
    output_dir="./attention_maps",
    save_every_n_epochs=10
)

trainer = pl.Trainer(callbacks=[visualizer])
```

### 🛠 Alternatives you've considered

- Manually extracting attention weights and creating custom visualizations
- Using external libraries like `pytorch-grad-cam` or `captum`
- Post-hoc analysis after training

These require significant boilerplate and aren't integrated with Lightly's pipeline.

### 📝 Additional context

**Use case:** Continuing pretraining DINOv2 on medical thermal imaging. Attention visualization would help verify the model isn't overfitting to artifacts and focuses on anatomically relevant regions.

This is especially valuable for continued pretraining where understanding what the model learns from domain-specific data is crucial for validation and debugging.

**Comment by stegmuel** [comment]
Hey,
Thanks for the suggestion. 
Based on my past experience, self-attention probing does not always correlate well with final performance.
For instance if you look at Fig. 14 from [DINOv3](https://arxiv.org/pdf/2508.10104), you will see that DINOv2 achieves lower CorLoc scores than DINO despite it performing much better in most "important" tasks. The CorLoc score is highly correlated with "convincing" self-attention maps.
Nonetheless, if we implement this feature, I'll let you know.

**Comment by liopeer** [comment]
Thanks for submitting the feature requrest @blackpearl006! I don't think we'll have the resources for this in the next few weeks, but we would definitely welcome a PR with that feature – and provide guidance – if you'd like to spend some time on it.

**Comment by blackpearl006** [comment]
@liopeer 

I'd like to take a crack at implementing this, starting with ViT-based models as a first step.

Could you provide some initial guidance? Specifically, I'm wondering about the best way to hook into the model to extract attention weights and if the proposed `pl.Trainer` callback API is the right approach.

Any pointers to relevant parts of the codebase would be much appreciated!

**Comment by liopeer** [comment]
I like the callback approach. That also keeps it nicely encapsulated and I have done similar things in the past in callbacks (in my case it was 3D renderings).

- The callback should sit in a separate file in `src/lightly_train/_callbacks`. You can check how the other callbacks in those files, e.g. `ModelCheckpoint` are implemented. 
- Together with the callback you should add an `Args` class that inherits from `from lightly_train._configs.config import PydanticConfig`. This ensures that all arguments passed to the callback are properly validated (since this will be an API that is exposed to the users).
- Afterwards make sure to add the `Args` class to `src/lightly_train/_callbacks/callback_args.py` into the `CallbackArgs` config.
- The actual `LightningModule` that will be passed into the callback is the `DINOv2` class from this file: `src/lightly_train/_methods/dinov2/dinov2.py`. So I think all the information required you should be able to find there.

Eventually we should make sure that this can only be used with the `"dinov2"` method, but this we can do at a later point.

When developing, make sure to follow the instructions in the `CONTRIBUTING.md`. In terms of code styling we also make sure to import classes directly:
```python
from something import SomeClass

SomeClass()
```
But for functions we always import the module:
```python
from something.something import something

something.somefunction()
```

Don't hesitate to ask questions, also over e.g. Discord (link in README) where we can respond a bit more real-time.

**Comment by liopeer** [comment]
One additional remark: In order for us to accept it it needs to support be able to run in distributed settings and should be covered by unit tests (unless the logic is super simple).

It's also usually best to ask for feedback early, so we can intervene if it doesn't go the direction we intended 🙂
